### ISSUE TITLE
[BUG] model.fit leads to TypeError 

### ISSUE THREAD

**Issue by jbottke** [basf/mamba-tabular#211]
**Describe the bug**
When trying to fit a classification model on the adult census dataset, the training process results in a TypeError while trying to concatenate an integer to a string in datamodule.py

**To Reproduce**
- load adult census dataset using shapiq with shapiq.datasets.load_adult_census()
- preprocess data using sklearns train_test_split() function which shuffles dataset und splits it into four numpy arrays (x_train, x_test, y_train, y_test)
- create Instance of MambularClassifier and fit it on numpy arrays x_train, y_train 

**Expected behavior**
Method is expected to further process training data and use it to train model. After that, the best model parameters are expected to be saved under model_checkpoints. 

**Screenshots**

![Image](https://github.com/user-attachments/assets/8a74cc90-6f1b-43c3-91db-b3afb0ca5289)

**Desktop:**
 - OS: Ubuntu via WSL2 
 - Python version 3.10.12
 - Mambular Version 1.1.0

**Comment by jbottke** [comment]
One is able to avoid the Problem by converting x_train from a numpy array to a pandas dataframe with named columns. It is important for the columns of the dataframe to be Strings. y_train can be a numpy array after all.
